web/src/stream_list.ts

```diff
@@ -291,6 +291,27 @@ function get_section_channel_plus_icon_url(section: StreamListSection): string |
     return undefined;
 }
 
+function maybe_change_channel_folders_option_visibility(): void {
+    let has_channel_folders = false;
+    const subscribed_subs = stream_data.subscribed_subs();
+
+    for (const sub of subscribed_subs) {
+        if (sub.folder_id) {
+            has_channel_folders = true;
+            break;
+        }
```

Comment:
This seems like it should be a little function we add: `channel_folders.user_has_folders()`.